corehq/apps/app_manager/views/utils.py

```diff
@@ -415,6 +415,37 @@ def form_has_submissions(domain, app_id, xmlns):
     return FormES().domain(domain).app([app_id]).xmlns([xmlns]).count() != 0
 
 
+def get_new_multimedia_between_builds(domain, target_build_id, source_build_id, build_profile_id=None):
+    source_build = get_app_cached(domain, source_build_id)
+    target_build = get_app_cached(domain, target_build_id)
+    assert source_build.copy_of, _("Size calculation available only for builds")
+    assert target_build.copy_of, _("Size calculation available only for builds")
+    build_profile = source_build.build_profiles[build_profile_id] if build_profile_id else None
+    multimedia_map_for_source_build_by_id = {
+        media_item['multimedia_id']: media_item
+        for path, media_item in
+        source_build.multimedia_map_for_build(build_profile=build_profile).items()
+    }
+    multimedia_map_for_target_build = target_build.multimedia_map_for_build(build_profile=build_profile)
+    multimedia_map_for_target_build_by_id = {
+        media_item['multimedia_id']: media_item
+        for path, media_item in
+        multimedia_map_for_target_build.items()
+    }
+    added = set(multimedia_map_for_target_build_by_id.keys()).difference(
+        set(multimedia_map_for_source_build_by_id.keys()))
```

Comment:
for keys that are common won't you also need to compare the MM version and if there version is different then also include that file in the total calculation?

Reply:
I believe the code is correct as-is. The multimedia id is associated with the content of a specific file. `media_suite.xml` needs to use a version because resource ids aren't multimedia ids, they're generated [based on the combination of multimedia_id and application path](https://github.com/dimagi/commcare-hq/blob/master/corehq/apps/hqmedia/models.py#L455) but they're [kept constant even when the file's content changes](https://github.com/dimagi/commcare-hq/blob/master/corehq/apps/app_manager/models.py#L4779-L4780), which makes version necessary.

I suspect we could simplify the media suite by using multimedia id as the resource id and ignoring the version altogether, but that's another conversation. For the purposes of this PR, I think using the multimedia id as a proxy for the file content is the right thing to do.

Docs here: https://github.com/dimagi/commcare-hq/tree/master/corehq/apps/hqmedia#the-apps-multimedia_map